src/components/settings/SafeAppsSigningMethod/index.test.tsx

```diff
@@ -0,0 +1,27 @@
+import { act, fireEvent, render } from '@/tests/test-utils'
+import { SafeAppsSigningMethod } from '.'
+
+describe('SafeAppsSigningMethod', () => {
+  it('Toggle on-chain signing', async () => {
+    const result = render(<SafeAppsSigningMethod />, {
+      initialReduxState: {
+        settings: {
+          signing: {
+            useOnChainSigning: false,
+          },
+        } as any,
+      },
+    })
+
+    const checkbox = result.getByRole('checkbox')
+    expect(checkbox).not.toBeChecked()
+
+    act(() => fireEvent.click(checkbox))
+
+    expect(checkbox).toBeChecked()
```

Comment:
can we also checked the Redux state is correct?

Reply:
Or maybe this test could start with `useOnChainSigning: true`

Reply:
I'm not sure if it is possible to access the redux state which we pass in to the `customRender`
I did not find any test which does that.